### PR TITLE
feat(flags): Shorter wait times before timing out

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

We'd have `FLAG_MATCHING_TIMEOUT_MS` for every request to the db when the first one timed out; which would push us over the 10s limit for multiple flags.

Similarly, having a 3s timeout on _every_ operation was too much, so have combined these limits together into a more logical grouping.

This still isn't the most optimal, as we can have in the worst case, 3s pgbouncer wait + 1s statement timeout everywhere it's called (5 places), so total 20s worst case, which is twice as long as the smallest we want.

However, to de-risk things, and know exactly what breaks, making this change in pieces.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/models/feature_flag/flag_matching.py [hunk 1]
```diff
@@ -245,85 +249,89 @@ def variant_lookup_table(self, feature_flag: FeatureFlag):
...
-                        target_properties = self.property_value_overrides
-                        if feature_flag.aggregation_group_type_index is not None:
-                            target_properties = self.group_property_value_overrides.get(
-                                self.cache.group_type_index_to_name[feature_flag.aggregation_group_type_index], {}
-                            )
-                        expr = properties_to_Q(
-                            Filter(data=condition).property_groups.flat,
-                            override_property_values=target_properties,
+        try:
+            with execute_with_timeout(FLAG_MATCHING_QUERY_TIMEOUT_MS):
```
Comment (neilkakkar) [inline posthog/models/feature_flag/flag_matching.py:253]: indent change only, try checking diff with whitespace off

**Comment by liyiy** [comment]
just wondering: Are the longer timeouts pretty much guaranteed failures too? Or even if they returned successfully we don't want the user to have to wait that long?

**Comment by neilkakkar** [comment]
>  Or even if they returned successfully we don't want the user to have to wait that long?

This, we're more concerned with returning a response quickly, since flags can be reloaded very easily, and do go through a few reloads ever so often in a single user's journey

**Review by EDsCODE — APPROVED** [review]
Nice work. One note, I had to string together your previous conversations in slack to get context as to what's going on and be able to think about this PR properly. Not really sure best fix here as I'm not suggesting you need to link the various threads as that would get tedious

**Comment by neilkakkar** [comment]
Yeah, I see how that could happen. This kind of work is surprisingly very iterative, need to tweak things till we get it just right

**Pull request merged**